src/test/HostActivationTests/FrameworkResolution/RollForwardReleaseOnly.cs

```diff
@@ -0,0 +1,222 @@
+        [InlineData(Constants.RollForwardSetting.Minor,       false,       "2.1.2")]
+        [InlineData(Constants.RollForwardSetting.LatestMinor, null,        "2.4.1")]
+        // applyPatches is ignored for new rollForward settings
+        [InlineData(Constants.RollForwardSetting.LatestMinor, false,       "2.4.1")]
+        [InlineData(Constants.RollForwardSetting.Major,       null,        "2.1.3")]
+        // Backward compat, equivalient to rollForwardOnNoCadidateFx=2, applyPatches=false
+        [InlineData(Constants.RollForwardSetting.Major,       false,       "2.1.2")]
+        [InlineData(Constants.RollForwardSetting.LatestMajor, null,        "3.1.2")]
+        public void RollForwardOnPatch_ReleaseOnly(string rollForward, bool? applyPatches, string resolvedFramework)
+        {
+            RunTest(
+                "2.1.0",
+                rollForward,
+                applyPatches)
+                .ShouldHaveResolvedFrameworkOrFail(MicrosoftNETCoreApp, resolvedFramework);
+        }
+
+        // Verifies that rollForward settings behave as expected when starting from 2.1.2 which does exist
+        // to other available 2.1.* versions. So roll forward on patch version.
+        [Theory] // rollForward                               applyPatches resolvedFramework
+        [InlineData(Constants.RollForwardSetting.Disable,     null,        "2.1.2")]
+        // applyPatches is ignored for new rollForward settings
+        [InlineData(Constants.RollForwardSetting.Disable,     false,       "2.1.2")]
+        // applyPatches is ignored for new rollForward settings
+        [InlineData(Constants.RollForwardSetting.Disable,     true,        "2.1.2")]
+        [InlineData(Constants.RollForwardSetting.LatestPatch, null,        "2.1.3")]
+        // Backward compat, equivalient to rollForwardOnNoCadidateFx=0, applyPatches=false
+        [InlineData(Constants.RollForwardSetting.LatestPatch, false,       "2.1.2")]
+        [InlineData(Constants.RollForwardSetting.Minor,       null,        "2.1.3")]
+        // Backward compat, equivalient to rollForwardOnNoCadidateFx=1, applyPatches=false
+        [InlineData(Constants.RollForwardSetting.Minor,       false,       "2.1.2")]
+        [InlineData(Constants.RollForwardSetting.Major,       null,        "2.1.3")]
+        // Backward compat, equivalient to rollForwardOnNoCadidateFx=2, applyPatches=false
+        [InlineData(Constants.RollForwardSetting.Major,       false,       "2.1.2")]
+        public void RollForwardOnPatch_FromExisting_ReleaseOnly(string rollForward, bool? applyPatches, string resolvedFramework)
+        {
+            RunTest(
+                "2.1.2",
+                rollForward,
+                applyPatches)
+                .ShouldHaveResolvedFrameworkOrFail(MicrosoftNETCoreApp, resolvedFramework);
+        }
+
+        // Verifies that rollForward settings behave as expected when starting from 2.0.0
```

Comment:
It would not be roll on minor. These tests are organized by "how much do we need to roll forward" - or maybe "how far off is the requested version from the closest available one".
Not perfect, but this is a matrix, so I picked one axis. This makes more sense in the old `rollForwardOnNoCandidateFx` where there are no weird `LatestMinor/LatestMajor`. In that world the resolved framework version is pretty much always the same.

I wrote the old tests first and these are basically just copies to verify that it works the same when using the new setting - and then I added the new options as well.